data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -1051,6 +1051,17 @@ impl TraceExporterBuilder {
             query_params: self.query_params,
         })
     }
+
+    fn is_mode_allowed(input: TraceExporterInputFormat, output: TraceExporterOutputFormat) -> bool {
```

Comment:
I think naming this something closer to `is_input_output_formats_compatible` is more descriptive. It will also make others think twice about just adding further validations to this function rather than a more maintainable solution.